multipart_http_signer.go

```diff
@@ -0,0 +1,232 @@
+package tests
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/pkg/randbytes"
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+func TestMultipartHTTPSignerCreateMultipart(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		signer, ok := store.(types.MultipartHTTPSigner)
+		So(ok, ShouldBeTrue)
+
+		Convey("When CreateMultipart via QuerySignHTTPCreateMultipart", func() {
+			path := uuid.New().String()
+			req, err := signer.QuerySignHTTPCreateMultipart(path, time.Duration(time.Hour))
+
+			Convey("The error should be nil", func() {
+				So(err, ShouldBeNil)
+
+				So(req, ShouldNotBeNil)
+				So(req.URL, ShouldNotBeNil)
+			})
+
+			client := http.Client{}
+			_, err = client.Do(req)
+
+			Convey("The request returned error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("List with ModePart should get the object without error", func() {
+				it, err := store.List(path, pairs.WithListMode(types.ListModePart))
+
+				So(err, ShouldBeNil)
+
+				o, err := it.Next()
+				So(err, ShouldBeNil)
+				So(o, ShouldNotBeNil)
+				So(o.Path, ShouldEqual, path)
+			})
+
+			defer func() {
+				it, err := store.List(path, pairs.WithListMode(types.ListModePart))
+				if err != nil {
+					t.Error(err)
+				}
+
+				o, err := it.Next()
+				if err != nil {
+					t.Error(err)
+				}
+
+				err = store.Delete(path, pairs.WithMultipartID(o.MustGetMultipartID()))
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+		})
+	})
+}
+
+func TestMultipartHTTPSignerWriteMultipart(t *testing.T, store types.Storager) {
```

Comment:
I wonder if we could combine these test cases into one method for `MultipartHTTPSigner` as usually these interfaces are all supported if presign supported?